code/modules/mob/living/carbon/xenomorph/abilities/predalien/predalien_powers.dm

```diff
@@ -42,116 +39,183 @@
 	apply_cooldown()
 	return ..()
 
-/datum/action/xeno_action/onclick/smash/use_ability(atom/target)
+/datum/action/xeno_action/activable/feralfrenzy/use_ability(atom/target)
 	var/mob/living/carbon/xenomorph/xeno = owner
+	if(!action_cooldown_check() || xeno.action_busy)
+		return
+	if(!xeno.check_state())
+		return
+
+	var/aoe_damage = base_damage_aoe
+	var/aoe_scale = damage_scale_aoe
+	var/range = 2
+	var/targetting_type = targetting
+
+	if(targetting_type == AOETARGETGUT)
+		var/datum/behavior_delegate/predalien_base/behavior = xeno.behavior_delegate
+		if(!istype(behavior))
+			return
+		if (range > 1)
+			xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] begins digging in for a massive strike!"), SPAN_XENOHIGHDANGER("We begin digging in for a massive strike!"))
+		else
+			xeno.visible_message(SPAN_XENODANGER("[xeno] begins digging in for a strike!"), SPAN_XENOHIGHDANGER("We begin digging in for a strike!"))
+
+		ADD_TRAIT(xeno, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Eviscerate"))
+		xeno.anchored = TRUE
+		if (do_after(xeno, (activation_delay_aoe), INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE))
+			xeno.emote("roar")
+			xeno.spin_circle()
+
+			for (var/mob/living/carbon/human in orange(xeno, range))
```

Comment:
Be careful with your variable naming here, your variable is named `human` but it could be any child of `mob/living/carbon` such as `xenomorph` this could confuse the next person to read your code.